ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestavt.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+import logging
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils.avdtestbase import AvdTestBase
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+LOGGER = logging.getLogger(__name__)
+
+
+class AvdTestAvtPath(AvdTestBase):
+    """
+    AvdTestAvtPath class for AVT (Adaptive Virtual Topology) tests.
+    Validates the status and type of AVT paths for a specified VRF in WAN scenarios.
+    It constructs a list of static peer addresses for each device by searching through
+    router_path_selection.path_groups.static_peers.
+    """
+
+    anta_module = "anta.tests.avt"
+
+    @cached_property
+    def test_definition(self) -> dict | None:
+        """
+        Generates the ANTA test definition for AVT path tests.
+
+        Returns:
+            dict | None: ANTA test definition or None if the configuration is incomplete.
+        """
+        # Retrieve path groups from the structured configuration
+        path_groups = get(self.structured_config, "router_path_selection.path_groups")
+        if not path_groups:
+            LOGGER.info("Path groups are not configured to collect AVT peer information. %s is skipped.", self.__class__.__name__)
+            return None
+
+        # Retrieve AVT profiles from the structured configuration
+        avt_profiles = get(self.structured_config, "router_adaptive_virtual_topology.vrfs")
```

Comment:
I am confused why vrf are called avt_profiles? under router_adaptive_virtual_topology you have a list of vrfs and under each vrf a list of profiles being listed right so not sure what we are picking up here

Reply:
updated the variable names and right now i am collecting all vrfs then under that all profiles and then making all possible commands for each destination as in structure config there no direct information about which destination is mapped with which vrf and avt profile